src/main/java/au/org/emii/netcdf/iterator/reader/NetcdfReader.java

```diff
@@ -76,4 +80,13 @@ public String getString(Set<IndexValue> indexValues) {
         }
     }
 
+    public String getDisplayValue(Set<IndexValue> indexValues) {
+        EnhanceScaleMissing scaleMissingDecorator = (EnhanceScaleMissing) variable;
+
+        if (scaleMissingDecorator.hasMissing() && scaleMissingDecorator.isMissing(getDouble(indexValues))) {
```

Comment:
Are they encoding NaN in the double?

Would it be better to have another property getMissing(), and then use that here.